File: exercise_sets_for_101_109/easy8/fizzbuzz.rb
def fizzbuzz(starting, ending)
  (1..15).each do |num|
    if num % 5 == 0 && num % 3 == 0
      puts "FizzBuzz"
    elsif num % 3 == 0
      puts "Fizz"
    elsif num % 5 == 0
      puts "Buzz"
    else
      puts num
    end
  end
end

def fizzbuzz2(starting, ending)
  result = []
  starting.upto(ending) do |number|
    result << fizzbuzz_value(number)
  end
  puts result.join(', ')
end

def fizzbuzz_value(number)
  case 
  when number % 3 == 0 && number % 5 == 0
    'FizzBuzz'
  when number % 5 == 0
    'Buzz'
  when number % 3 == 0
    'Fizz'
  else 
    number
  end
end

fizzbuzz(1,15)
fizzbuzz2(1,15)

File: 210/lesson3/fizzbuzz.js
function fizzbuzz() {
  var message;

  for (var i = 1; i <= 100; i++) {
    message = i;

    if (i % 3 === 0 && i % 5 === 0) {
      message = "FizzBuzz";
    } else if (i % 5 === 0) {
      message = "Buzz";
    } else if (i % 3 === 0) {
      message = "Fizz";
    } 

    console.log(message);
  }
}

function fizzbuzz() {
  var message;

  for (var i = 1; i <= 100; i++) {
    message = '';

    if (i % 3 === 0) {
      message += 'Fizz'
    }

    if (i % 5 === 0) {
      message += 'Buzz'
    }

    console.log(message || i);
  }
}

fizzbuzz()